app/viewmodels/checkAnswers/AnyOtherIncomeSummary.scala

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package viewmodels.checkAnswers
+
+import controllers.journeys.income.routes.AnyOtherIncomeController
+import models.{CheckMode, UserAnswers}
+import pages.AnyOtherIncomePage
+import play.api.i18n.Messages
+import uk.gov.hmrc.govukfrontend.views.viewmodels.summarylist.SummaryListRow
+import viewmodels.govuk.summarylist._
+import viewmodels.implicits._
+
+object AnyOtherIncomeSummary {
+
+  def row(answers: UserAnswers, taxYear: Int)(implicit messages: Messages): Option[SummaryListRow] =
```

Comment:
Me and Hassan were looking at the way we generate the `SummaryListRow` in the CYA view models, and we were thinking it would make sense to handle the `None` case (where no user answers are present for the `AnyOtherIncomePage`) inside this method, instead of the exception being generated in the controller itself. See [here](https://github.com/hmrc/income-tax-self-employment-frontend/pull/16/files#diff-304f6b650510e6effffaaf23b9125050da29d49161aa983538b5839909920f26R29-R44) 

What are your thoughts?

Reply:
Some summary rows need to be optional and the user answers from most pages will arrive not in an Option after ticket SASS-5841 is implemented.
For that reason the summary models can be left as is or can be edited on a case by case bases in the way you have described when ticket SASS-5842 is being done. I shall leave a link to this comment in ticket 5842.